### PR TITLE
Astroid 2.5.0

### PR DESCRIPTION
**Pull request by hippo91**
## Steps

- [x] Write a good description on what the PR does.

## Description
This is a PR for future 2.5.0 version.
When merged we wiil have to do the 4-5 and 6 steps of the release.txt file.

### CODE REVIEW

**Comment by hippo91** [comment]
@PCManticore have you got an idea on how to put a deprecation warning when `astroid` is used with `python3.5`?

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
@hippo91, regarding the deprecation warning I would use the following in `setup.py`:

```
import sys
import warnings

if sys.version_info.major == 3 and sys.version_info.minor <=5:
    warnings.warn("You will soon need to upgrade to python 3.6 in order to use the latest version of Astroid.", DeprecationWarning)
```

**Comment by hippo91** [comment]
@Pierre-Sassoulas definitely a good idea! The message will be printed during install and not at runtime. It's probably sufficient.

File: ChangeLog [hunk 1]
```diff
@@ -5,8 +5,7 @@ astroid's ChangeLog
 
 What's New in astroid 2.5.0?
 ============================
-Release Date: TBA
-
+Release Date: 2020-12-27
```
Comment (PCManticore) [inline ChangeLog:8]: This is already obsolete. My suggestion is to simply backport the code changes but keep the `__pkginfo__` and `ChangeLog` for a separate change in the day of the release.

File: astroid/modutils.py [hunk 2]
```diff
@@ -89,11 +91,23 @@
             pass
 
 if platform.python_implementation() == "PyPy":
+    #  The get_python_lib(standard_lib=True) function does not give valid
```
Comment (PCManticore) [inline astroid/modutils.py:94]: Fantastic!

**Review by PCManticore — APPROVED** [review]
Approved, but ideally we should remove or change the release date to today.

File: ChangeLog (hunk 1 above)
Comment (hippo91) [reply ChangeLog:8]: @PCManticore i will move back the Release Date in the ChangeLog. However i didn't change the `__pkginfo__`.
By the way, have you an idea concerning the release date? There are many people waiting for a release that support python 3.9.

Comment (PCManticore) [reply ChangeLog]: Thanks. Not right now. I am still going through the latest updates, mails and issues and don't have a clear picture on what's needed for Python 3.9 compatibility. I did notice some Python 3.9 issues specific to `pylint` though.

**Comment by FRidh** [comment]
Could there be a tag and PyPI release?

**Pull request merged**